tests/integration/config/plugins/integration-tests/data/tests/tests_spawn_limits.txt

```diff
@@ -0,0 +1,254 @@
+					limit 0
+					initial count 1
+					ignore enemy strength
+				fleet "Disabled 2 Test Fleet"
+					period 100000000
+					limit 0
+					initial count 1
+
+
+
+fleet "Non-Disabled A Test Fleet"
+	government "Test Government 1 A"
+	cargo 0
+	personality staying heroic
+	variant 60
+		"Aphid"
+
+fleet "Disabled B Test Fleet"
+	government "Test Government 1 B"
+	cargo 0
+	personality staying derelict heroic
+	variant 60
+		"Star Barge"
+
+fleet "Disabled C Test Fleet"
+	government "Test Government 1 C"
+	cargo 0
+	personality staying derelict heroic
+	variant 60
+		"Shuttle"
+
+government "Test Government 1 A"
+	swizzle 2
+	color 1 1 1
+	"player reputation" 1e9
+	"fine" 0
+	"attitude toward"
+		"Test Government 2" -1
+
+government "Test Government 1 B"
+	swizzle 2
+	color 1 1 1
+	"player reputation" 1e9
+	"fine" 0
+	"attitude toward"
+		"Test Government 2" -1
+
+government "Test Government 1 C"
+	swizzle 3
+	color 1 1 1
+	"player reputation" 1e9
+	"fine" 0
+	"attitude toward"
+		"Test Government 2" -1
+
+government "Test Government 2"
+	swizzle 4
+	color 1 1 1
+	"player reputation" 1e9
+	"fine" 0
+	"attitude toward"
+		"Test Government 1 A" -1
+		"Test Government 1 B" -1
+		"Test Government 1 C" -1
+
+mission "Board 1 A"
+	assisting
+	repeat
+	source
+		government "Test Government 1 A"
+	on offer
+		"test: Test Government 1 A boarded" ++
+	destination "Earth"
+
+mission "Board 1 B"
+	assisting
+	repeat
+	source
+		government "Test Government 1 B"
+	on offer
+		"test: Test Government 1 B boarded" ++
+	destination "Earth"
+
+mission "Board 1 C"
+	assisting
+	repeat
+	source
+		government "Test Government 1 C"
+	on offer
+		"test: Test Government 1 C boarded" ++
+	destination "Earth"
+
+mission "Board 2"
+	assisting
+	repeat
+	source
+		government "Test Government 2"
+	on offer
+		"test: Test Government 2 boarded" ++
+	destination "Earth"
+
+ship "Kasichara A'awoj" "Big Ship"
+	outfits
+		"Triple Plasma Core"
+		"Systems Core (Large)"
+		"Systems Core (Small)"
+		"Large Heat Shunt" 2
+		"Small Heat Shunt" 3
+		"Fuel Processor" 2
+		"Luxury Accommodations"
+		"Microbot Defense Station" 23
+		"Thermal Repeater Rifle" 548
+
+fleet "Disabled 2 Test Fleet"
+	government "Test Government 2"
+	cargo 0
+	personality staying derelict
+	variant 60
+		"Big Ship"
+
+
+
+test "Spawn Limit Tests"
+	status active
+	description "Tests if random interval and system entry fleet spawn limits work."
+	sequence
+		inject "Spawn Limits Test"
+		call "Load First Savegame"
+		watchdog 12000
+		call "Depart"
+		# Wait for 10 seconds to drift away from the planet.
+		apply
+			"test: steps to wait" = 10 * 60
+		# empty input to make time pass
+		input
+
+		# Board first ship.
+		input
+			command board
+		apply
+			"test: steps to wait" = 3 * 60
+		label wait1
+		apply
+			"test: steps to wait" = "test: steps to wait" - 1
+		branch wait1
+			"test: steps to wait" > 0
+
+		# Board second ship.
+		input
+			command board
+		apply
+			"test: steps to wait" = 3 * 60
```

Comment:
Why do you skip a given amoint of time when you could just check if the boarding mission was offered?
You should add a watchdog too

Reply:
Don't I need to wait for the ship to arrive, and the pilot confusion to expire before sending the board command again? Otherwise, it'll just go to the same ship.

Reply:
I'm confused what you're doing here anyway
Why not just make the mission blow the ship off once you're done, using a launch endpoint?

Reply:
Then another ship will appear because a limit is no longer reached (number of fleets decreased by 1).

Reply:
I will add a comment explaining how the test works. It has to be designed quite precisely to test the different spawn rate limits.